clients/src/main/java/org/apache/kafka/common/metrics/JmxReporter.java

```diff
@@ -49,7 +49,7 @@
     private static final Logger log = LoggerFactory.getLogger(JmxReporter.class);
     private static final Object LOCK = new Object();
     private String prefix;
-    private final Map<String, KafkaMbean> mbeans = new HashMap<String, KafkaMbean>();
+    final Map<String, KafkaMbean> mbeans = new HashMap<String, KafkaMbean>();
```

Comment:
We may not want to expose this map to outside of this class as it can be modified using put/remove. Oneway to address is adding a method returning unmodifiable map. But it may be good to restrict with package level access method like `containsMBean(String mbeanName)` as the required test needs to know whether a mbean with a name exists or not.